src/systems/wheel_slip/WheelSlip.cc

```diff
@@ -0,0 +1,367 @@
+            << "] in model [" << modelName << "]"
+            << std::endl;
+      continue;
+    }
+
+    if (params.wheelNormalForce <= 0)
+    {
+      ignerr << "Found wheel normal force [" << params.wheelNormalForce
+            << "], which is not positive"
+            << " in link named [" << linkName
+            << "] in model [" << modelName << "]"
+            << std::endl;
+      continue;
+    }
+
+    this->mapLinkSurfaceParams[link.Entity()] = params;
+  }
+
+  if (this->mapLinkSurfaceParams.empty())
+  {
+    ignerr << "No links and surfaces found, plugin is disabled"
+           << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+/////////////////////////////////////////////////
+void WheelSlipPrivate::Update(EntityComponentManager &_ecm)
+{
+  for (const auto &linkSurface : this->mapLinkSurfaceParams)
+  {
+    const auto &params = linkSurface.second;
+
+    // get user-defined normal force constant
+    double force = params.wheelNormalForce;
+
+    auto joint = params.joint;
+
+    auto spinAngularVelocityComp =
+        _ecm.Component<components::JointVelocity>(joint);
+
+    if (!spinAngularVelocityComp || spinAngularVelocityComp->Data().empty())
+      continue;
+    double spinAngularVelocity = spinAngularVelocityComp->Data()[0];
+
+    // As discussed in WheelSlip.hh, the slip1 and slip2
+    // parameters have units of inverse viscous damping:
+    // [linear velocity / force] or [m / s / N].
+    // Since the slip compliance parameters supplied to the plugin
+    // are unitless, they must be scaled by a linear speed and force
+    // magnitude.
+    // The force is taken from a user-defined constant that should roughly
+    // match the steady-state normal force at the wheel.
+    // The linear speed is computed dynamically at each time step as
+    // radius * spin angular velocity.
+    // This choice of linear speed corresponds to the denominator of
+    // the slip ratio during acceleration (see equation (1) in
+    // Yoshida, Hamano 2002 DOI 10.1109/ROBOT.2002.1013712
+    // "Motion dynamics of a rover with slip-based traction model").
+    // The acceleration form is more well-behaved numerically at low-speed
+    // and when the vehicle is at rest than the braking form,
+    // so it is used for both slip directions.
+    double speed = params.wheelRadius * std::abs(spinAngularVelocity);
+    double slip1 = speed / force * params.slipComplianceLateral;
+    double slip2 = speed / force * params.slipComplianceLongitudinal;
+
+    components::SlipComplianceCmd newSlipCmdComp({slip1, slip2});
+
+    auto currSlipCmdComp =
+        _ecm.Component<components::SlipComplianceCmd>(params.collision);
+    if (currSlipCmdComp)
+    {
+      auto state =
+        currSlipCmdComp->SetData(newSlipCmdComp.Data(), this->vecEql) ?
+        ComponentState::OneTimeChange :
+        ComponentState::NoChange;
+      _ecm.SetChanged(params.collision,
+          components::SlipComplianceCmd::typeId, state);
```

Comment:
I noticed that setting the `SlipComplianceCmd` component this way causes simulation to slow down significantly. My RTF goes from 100% to ~70% when I start driving a robot with WheelSlip enabled. I recommend using the `PeriodicChange` state to avoid the serialization overhead.

```suggestion
      *currSlipCmdComp = newSlipCmdComp;
      _ecm.SetChanged(params.collision, components::SlipComplianceCmd::typeId,
                      ComponentState::PeriodicChange);
```